sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client_async.py

```diff
@@ -124,7 +126,21 @@ def _create_pipeline(self, credential, **kwargs):
         try:
             if response.status_code not in [202]:
                 raise HttpResponseError(response=response)
-            return response.parts()  # Return an AsyncIterator
+            parts = response.parts() # Return an AsyncIterator
+            parts_list = []
+            if raise_on_any_failure:
+                failures = []
+                async for part in parts:
+                    parts_list.append(part)
+                    if not (200 <= part.status_code < 300):
+                        failures.append(part)
+                if failures:
+                    error = PartialBatchErrorException(
+                        message="There is a partial failure in the batch operation.",
+                        response=response, parts=AsyncList(parts_list)
+                    )
+                    raise error
+            return AsyncList(parts_list)
```

Comment:
`parts_list` is empty if `raise_on_any_failure` is False, is that intentional?
I think you could update to:
```
parts_list = []
failures = []
async for part in parts:
    parts_list.append(part)
    if not 200 <= part.status_code < 300:
        failures.append(part)
if raise_on_any_failure and failures:
    error = PartialBatchErrorException(
        message="There is a partial failure in the batch operation.",
        response=response, parts=AsyncList(parts_list)
    )
    raise error
return AsyncList(parts_list)
```

Reply:
fixed